docs/data/base/components/number-input/NumberInputAdornments.tsx

```diff
@@ -0,0 +1,189 @@
+import * as React from 'react';
+import {
+  Unstable_NumberInput as NumberInput,
+  NumberInputProps,
+  numberInputClasses,
+} from '@mui/base/Unstable_NumberInput';
+import Box from '@mui/material/Box';
+import { styled } from '@mui/system';
+
+const CustomNumberInput = React.forwardRef(function CustomNumberInput(
+  props: NumberInputProps,
+  ref: React.ForwardedRef<HTMLDivElement>,
+) {
+  return (
+    <NumberInput
+      slots={{
+        root: StyledInputRoot,
+        input: StyledInputElement,
+        incrementButton: StyledButton,
+        decrementButton: StyledButton,
+      }}
+      slotProps={{
+        incrementButton: {
+          children: <span className="arrow">▴</span>,
+        },
+        decrementButton: {
+          children: <span className="arrow">▾</span>,
+        },
+      }}
+      {...props}
+      ref={ref}
+    />
+  );
+});
+
+export default function NumberInputAdornments() {
+  return (
+    <Box
+      sx={{
+        display: 'flex',
+        flexDirection: { xs: 'column', sm: 'row' },
+        gap: 2,
+      }}
+    >
+      <CustomNumberInput
+        startAdornment={
+          <InputAdornment>
+            <svg
+              // From Feather: https://feathericons.com/?query=dollar-sign
```

Comment:
I swapped the icon so the demo doesn't need to import from `icons-material` and also to look less Material Design 😬